abi-spec/types.go

```diff
@@ -0,0 +1,190 @@
+package abispec
```

Comment:
could you elaborate on why types needed ?

Reply:
`types` is the `bridge` between solidity type and go type, we need go type to invoke abi functions in go-ethereum @flexsurfer